Makefile

```diff
@@ -29,6 +34,50 @@ test:
 	@echo "running against `git version`"; \
 	$(GOTEST) -race ./...
 
+TEST_ENV=.env
+DOCKER_ENV=.env-docker
+DOCKER_GOGIT_KEY = go-git
+DOCKER_GOGIT_LABEL = testing
+DOCKER_GOGIT_NAME = $(DOCKER_GOGIT_KEY)-dist
+.PHONY: test-env test-go1.19 test-go1.20 test-go1.21 test-go-all
+test-env:
+	cat <<EOF >$(TEST_ENV)
+	DOCKER_ENV=$(DOCKER_ENV)
+	DOCKER_GOGIT_KEY=$(DOCKER_GOGIT_KEY)
+	DOCKER_GOGIT_LABEL=$(DOCKER_GOGIT_LABEL)
+	DOCKER_GOGIT_NAME=$(DOCKER_GOGIT_NAME)
+	GIT_REPOSITORY=$(GIT_REPOSITORY)
+	GIT_VERSIONS=(master v2.11.0)
+	GO_VERSION_WGIT=1.21
+	GO_VERSIONS=(1.19 1.20 1.21)
+	WORKDIR=$(WORKDIR)
+	EOF
+
+test-go1.19: test-env
+	export BASH_ENV=$(TEST_ENV)
+	$(BASHCMD) _local/test-go.sh "1.19"
+
+test-go1.20: test-env
+	export BASH_ENV=$(TEST_ENV)
+	$(BASHCMD) _local/test-go.sh "1.20"
+
+test-go1.21: test-env
+	BASH_ENV=$(TEST_ENV)
+	$(BASHCMD) _local/test-go.sh "1.21"
+
+test-go-all: test-go1.19 test-go1.20 test-go1.21
+
+.PHONY: test-git-master test-git-v2.11
+test-git-v2.11:
+	BASH_ENV=$(TEST_ENV)
+	$(BASHCMD) _local/test-git-compatability.sh "v2.11.0"
+
+test-git-master:
+	BASH_ENV=$(TEST_ENV)
+	$(BASHCMD) _local/test-git-compatability.sh "master"
```

Comment:
Same here, let's have a single target and source the branch name from an environment var.